### PR TITLE
[FE-15] Groups list filtering

### PR DESCRIPTION
**Pull request by AnCichocka**
https://oops2000.atlassian.net/browse/SCRUM-15

- add sidebar filtering
- add tabs

image

### CODE REVIEW

File: frontend/src/hooks/Groups/useGroupsData.tsx [hunk 1]
```diff
@@ -24,7 +29,8 @@ export const useGroupsData = () => {
       return {
         name: group.groupName,
         id: group.groupsId,
-        weekday: group.weekday,
+        // TODO fix
+        weekday: group.weekday.toUpperCase(),
```
Comment (AnCichocka) [inline frontend/src/hooks/Groups/useGroupsData.tsx:33]: @Szyntos `frontend/src/grapql/groups.graphl` returns weekday lowercase and `frontend/src/grapql/.graphl` returns weekday uppercase - this inconsistency should not happen

File: frontend/src/graphql/weekdays.graphql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+query weekdays($editionId: Int!) {
+  getPossibleGroupsWeekdays(editionId: $editionId)
+}
```
Comment (AnCichocka) [inline frontend/src/graphql/weekdays.graphql:3]: We should add sorting on the backend so weekdays are not in random sequence - it should go from Monday to Friday. I we should probably create another query which returns all days of week and add filed "disabled" - for example, when there are no groups on Mondays it should be disabled. I find it confusing not seeing some weekdays knowing, what do you think? (not necessary for this PR)

Comment (Szyntos) [reply frontend/src/graphql/weekdays.graphql:3]: Done in [this PR](https://github.com/Jullija/oops/pull/78)

File: frontend/src/hooks/Groups/useGroupsData.tsx (hunk 1 above)
Comment (Szyntos) [reply frontend/src/hooks/Groups/useGroupsData.tsx:33]: also done in [this PR](https://github.com/Jullija/oops/pull/78)

**Review by Szyntos — APPROVED** [review]
Few small comments

This looks pretty already!

**Comment by Szyntos** [comment]
BTW shouldn't we add searching by the date and hour? Just to have unified flow?

**Comment by Szyntos** [comment]
I think the buttons should activate even if you click on the text.

File: frontend/src/screens/Groups/Groups.tsx [hunk 3]
```diff
@@ -1,25 +1,149 @@
...
-
   return (
-    <div>
-      <GroupSearchField onInputChange={onInputChange} groups={groups} />
-      <GroupsList groups={filteredGroups ?? groups} />
+    <div style={styles.container}>
+      <SideFilterBar
+        sections={[
+          {
+            pickerTitle: "dzień tygodnia",
```
[the remark this answers is not in the document]
Comment (AnCichocka) [reply frontend/src/screens/Groups/Groups.tsx:118]: What should it be then?

[the remark this answers is not in the document]
Comment (AnCichocka) [reply frontend/src/screens/Groups/Groups.tsx:118]: I will correct it, but these are details that are irrelevant now - I believe we will make one big correction on the end

**Comment by AnCichocka** [comment]
> BTW shouldn't we add searching by the date and hour? Just to have unified flow?

I don't like this idea, tbh I think we should stick to search only by group name - I think user might not understand results of searching

File: frontend/src/graphql/teachers.graphql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+      _and: {
+        role: { _in: ["teacher", "coordinator"] }
+        userGroups: { group: { editionId: { _eq: $editionId } } }
+      }
+    }
+  ) {
+    userId
+    fullName
+  }
+}
```
[the remark this answers is not in the document]
Comment (AnCichocka) [reply frontend/src/graphql/teachers.graphql:13]: but I need all teachers from edition, not teacher for group

**Pull request merged**